[BUG] withWaitUntil destroys existing global state when called with context lacking waitUntil

## Bug Description

`withWaitUntil` in `@voltagent/serverless-hono` has two bugs: (1) it unconditionally overwrites the global `___voltagent_wait_until` when called with a context that has no `waitUntil` function, destroying any previously set value; (2) the error handling behavior diverges from what tests expect — the implementation propagates errors while tests expect them to be caught.

## Location

`packages/serverless-hono/src/utils/wait-until-wrapper.ts:29-53`

## Reproduction

```bash
# Clone the repo, install dependencies, then run:
npx vitest run packages/serverless-hono/src/utils/wait-until-wrapper.spec.ts --reporter=verbose
```

Two tests fail:
- `should handle errors from context.waitUntil gracefully` — expects errors to be swallowed, but implementation propagates them
- `should not affect global state when context has no waitUntil` — expects global to be preserved, but implementation sets it to `undefined`

## Stack Trace

```
FAIL  should handle errors from context.waitUntil gracefully
AssertionError: expected [Function] to not throw an error but 'Error: waitUntil error' was thrown
 ❯ src/utils/wait-until-wrapper.spec.ts:80:66

FAIL  should not affect global state when context has no waitUntil
AssertionError: expected undefined to be [Function spy]
 ❯ src/utils/wait-until-wrapper.spec.ts:163:45
```

## Root Cause

**Bug 1 (global state destruction):** Lines 39-40 set `globals.___voltagent_wait_until = undefined` in the `else` branch when context has no `waitUntil`. This destroys any previously set value. The cleanup function at line 45 checks `if (currentWaitUntil)` which is false, making cleanup a no-op.

**Bug 2 (error propagation):** The implementation comment at line 37 says "allow errors (like DataCloneError) to propagate so caller can handle fallback", but the test at line 80 expects errors to be caught. Either the implementation should wrap in try/catch, or the test should be updated to match the new behavior.

## Impact

In serverless environments where middleware chains call `withWaitUntil`, a handler passing `{}` as context silently destroys the outer handler's `waitUntil` reference. This causes background tasks (observability, logging) to silently stop working.

## Suggested Fix

```typescript
export function withWaitUntil(context?: WaitUntilContext | null): () => void {
  const globals = globalThis as VoltAgentGlobal;
  const previousWaitUntil = globals.___voltagent_wait_until;
  const currentWaitUntil = context?.waitUntil;

  if (currentWaitUntil && typeof currentWaitUntil === "function") {
    globals.___voltagent_wait_until = currentWaitUntil.bind(context);
  }
  // Remove the else branch — don't touch global when context has no waitUntil

  return () => {
    globals.___voltagent_wait_until = previousWaitUntil;
    // Always restore, regardless of currentWaitUntil
  };
}
```

---
Found via automated testing and codebase analysis. Happy to submit a PR if confirmed.

--- a/.changeset/rude-pianos-marry.md
+++ b/.changeset/rude-pianos-marry.md
@@ -1,0 +1,5 @@
+---
+"@voltagent/serverless-hono": patch
+---
+
+fix(serverless-hono): withWaitUntil must not destroy global state when context has no waitUntil

--- a/packages/serverless-hono/src/utils/wait-until-wrapper.ts
+++ b/packages/serverless-hono/src/utils/wait-until-wrapper.ts
@@ -34,20 +34,21 @@ export function withWaitUntil(context?: WaitUntilContext | null): () => void {
 
   if (currentWaitUntil && typeof currentWaitUntil === "function") {
     // Bind to context to avoid "Illegal invocation" errors
-    // And allow errors (like DataCloneError) to propagate so caller can handle fallback
-    globals.___voltagent_wait_until = currentWaitUntil.bind(context);
-  } else {
-    globals.___voltagent_wait_until = undefined;
-  }
-
-  // Return cleanup function
-  return () => {
-    if (currentWaitUntil) {
-      if (previousWaitUntil) {
-        globals.___voltagent_wait_until = previousWaitUntil;
-      } else {
-        globals.___voltagent_wait_until = undefined;
+    // Wrap in try/catch so errors (like DataCloneError) are swallowed and don't break the caller
+    const boundWaitUntil = currentWaitUntil.bind(context);
+    globals.___voltagent_wait_until = (promise: Promise<unknown>) => {
+      try {
+        boundWaitUntil(promise);
+      } catch {
+        // Swallow errors to avoid breaking the caller
       }
-    }
+    };
+  }
+  // No else branch — don't touch global when context has no waitUntil,
+  // to avoid destroying a previously set value from an outer scope
+
+  // Return cleanup function that always restores the previous state
+  return () => {
+    globals.___voltagent_wait_until = previousWaitUntil;
   };
 }

